releso/callback.py

```diff
@@ -145,3 +145,112 @@ def _on_step(self) -> bool:
     def _on_training_end(self) -> None:
         """Function is called when training is terminated."""
         self._export()
+
+
+class StepInformationLogCallback(BaseCallback):
+    """Step Callback class.
+    
+    This class tracks all step-wise information that might come in handy 
+    during evaluation. Originally created to easily track the actions 
+    undertaken in each episode for better evaluation of the learned policy.
+    """
+
+    def __init__(
+        self,
+        step_log_location: Path,
+        verbose: int = 0,
+        update_every: Union[str,int] = "episode",
+    ):
+        """Constructor for the Callback using SB3 interface.
+
+        Args:
+            step_log_location (Path): Path to the step log file.
+            verbose (int, optional): Verbosity of the callback. Defaults to 0.
+            update_every (Union[str|int], optional): Update the step log file
+            every n steps. Default is after every episode.
+        """
+        super().__init__(verbose)
+
+        self.step_log_location: Path = step_log_location 
+        self.current_episode: int = 0
+        self.update_every: str = update_every

```

Comment:
But maybe it should here already be changed to only int.
```suggestion
        self.update_frequency: int = update_every if isinstance(update_every, int) else -1
```

Otherwise, you have a `string` or `int` variable here and mirror it later on with another variable though then it is only int.